src/shared/schemas.ts

```diff
@@ -183,43 +183,39 @@ export async function getDefaultSchemas(extensionContext: vscode.ExtensionContex
  * @param params.cacheKey Cache key to check version against
  * @param params.extensionContext VSCode extension context
  */
-export async function getRemoteOrCachedFile(params: {
-    filepath: string
+export async function updateSchemaFromRemote(params: {
+    destination: vscode.Uri
     version?: string
     url: string
     cacheKey: string
     extensionContext: vscode.ExtensionContext
-}): Promise<string> {
-    const dir = path.parse(params.filepath).dir
-    if (!(await filesystemUtilities.fileExists(dir))) {
-        mkdirSync(dir, { recursive: true })
-    }
-
+    title: string
+}): Promise<void> {
     const cachedVersion = params.extensionContext.globalState.get<string>(params.cacheKey)
     const outdated = params.version && params.version !== cachedVersion
     if (!outdated) {
         // Check that the cached file actually can be fetched. Else we might
         // never update the cache.
-        const fileFetcher = new FileResourceFetcher(params.filepath)
+        const fileFetcher = new FileResourceFetcher(params.destination.fsPath)
         const content = await fileFetcher.get()
         if (content) {
-            return content
+            return
         }
     }
 
-    const httpFetcher = new HttpResourceFetcher(params.url, {
-        showUrl: true,
-        // updates curr version
-        onSuccess: contents => {
-            writeFileSync(params.filepath, contents)
-            params.extensionContext.globalState.update(params.cacheKey, params.version)
-        },
-    })
+    const httpFetcher = new HttpResourceFetcher(params.url, { showUrl: true })
```

Comment:
I don't see anything that would regress, but to be sure, please check the history of this file and check that it didn't regress this fix: 2609b63c877e

That fix is not covered by tests.

Reply:
Yeah I already looked at the original PR + tested manually.

Reply:
Could be made a bit more robust though. It should be using the cached content if it fails to get the latest version.

Reply:
> It should be using the cached content if it fails to get the latest version.

yes--if it doesn't do that it's a regression